src/components/common/modal/bulk-edit-modal.tsx

```diff
@@ -0,0 +1,102 @@
+import { yupResolver } from "@hookform/resolvers/yup";
+import { Form } from "@lifesg/react-design-system/form";
+import { Text } from "@lifesg/react-design-system/text";
+import { Controller, FormProvider, useForm } from "react-hook-form";
+import { IBulkEditModalProps } from "src/context-providers";
+import { BULK_EDIT_SCHEMA, TBulkEditData } from "src/schemas/bulk-edit";
+import { useModal } from "../../../context-providers/display/modal-hook";
+import { Wrapper } from "./bulk-edit-modal.styles";
+import { ActionButton, ButtonWrapper } from "./discard-changes-modal.styles";
+import { GenericModal } from "./generic-modal";
+
+interface IProps {
+    modal: IBulkEditModalProps;
+}
+
+export const BulkEditModal = ({ modal }: IProps) => {
+    // =========================================================================
+    // CONST, STATE, REFS
+    // =========================================================================
+    const { hideModal } = useModal();
+    const { type, onClickActionButton, dropdownItemsString } = modal;
+    const methods = useForm({
+        mode: "onTouched",
+        resolver: yupResolver(BULK_EDIT_SCHEMA),
+        defaultValues: {
+            items: dropdownItemsString,
+        },
+    });
+    // ============================================================================
+    // HELPER FUNCTIONS
+    // ============================================================================
+
+    const handleOnClickCancel = () => {
+        hideModal(type);
+    };
+
+    const onSubmit = (data: TBulkEditData) => {
```

Comment:
prefix with `handle` for event handlers

```suggestion
    const handleFormSubmit = (data: TBulkEditData) => {
```